File: rb/features.rb
# BlutvAuthentication SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BlutvAuthenticationFeatures
  def self.make_feature(name)
    case name
    when "base"
      BlutvAuthenticationBaseFeature.new
    when "test"
      BlutvAuthenticationTestFeature.new
    else
      BlutvAuthenticationBaseFeature.new
    end
  end
end

File: py/features.py
# BlutvAuthentication SDK feature factory

from feature.base_feature import BlutvAuthenticationBaseFeature
from feature.test_feature import BlutvAuthenticationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BlutvAuthenticationBaseFeature(),
        "test": lambda: BlutvAuthenticationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
